onnx/optimizer/passes/fuse_matmul_add_bias_into_gemm.h

```diff
@@ -0,0 +1,94 @@
+// ATTENTION: The code in this file is highly EXPERIMENTAL.
+// Adventurous users should note that the APIs will probably change.
+
+#pragma once
+
+// Before:
+//   Z = MatMul(X, Y)
+//   B = Z + A
+// After:
+//   B = Gemm(X, Y, A)
+//
+// the pass can handle the case when A is 1D tensor and A.dim[0] == Z.dim[1]
+
+#include <numeric>
+
+#include "onnx/common/assertions.h"
+#include "onnx/optimizer/pass.h"
+
+namespace ONNX_NAMESPACE {
+namespace optimization {
+
+struct FuseMatMulAddBiasIntoGemm final : public PredicateBasedPass {
+  explicit FuseMatMulAddBiasIntoGemm()
+      : PredicateBasedPass(
+            PassType::Fuse,
+            PassEfficiency::Complete,
+            PassOptimizationType::Compute) {}
+  std::string getPassName() const override {
+    return "fuse_matmul_add_bias_into_gemm";
+  }
+  bool patternMatchPredicate(Node* node) override {
+    return node->kind() == kAdd &&
+        strcmp(node->inputs()[0]->node()->kind().toString(), "MatMul") == 0 &&
```

Comment:
Instead of doing a string comparison what you should do is add MatMul to interned_strings.h. Here's an example with the dropout op: https://github.com/onnx/onnx/pull/1555/files

Reply:
Thanks Armen, I was wondering if I should do it that way, but I wanted to ask first. I'll make the change.